packages/fx-core/src/common/cpUtils.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { LogProvider } from "@microsoft/teamsfx-api";
+import * as cp from "child_process";
+import * as os from "os";
+
+export async function executeCommand(
+  workingDirectory: string | undefined,
+  logger: LogProvider | undefined,
+  options: cp.SpawnOptions | undefined,
+  command: string,
+  ...args: string[]
+): Promise<string> {
+  const result: ICommandResult = await tryExecuteCommand(
+    workingDirectory,
+    logger,
+    options,
+    command,
+    ...args
+  );
+  if (result.code !== 0) {
+    const errorMessage = `Failed to run command: "${command} ${result.formattedArgs}", code: "${result.code}",
+                            output: "${result.cmdOutput}", error: "${result.cmdOutputIncludingStderr}"`;
+    await logger?.debug(errorMessage);
+    throw new Error(errorMessage);
+  } else {
+    await logger?.debug(`Finished running command: "${command} ${result.formattedArgs}".`);
+  }
+
+  return result.cmdOutput;
+}
+
+async function tryExecuteCommand(
+  workingDirectory: string | undefined,
+  logger: LogProvider | undefined,
+  additionalOptions: cp.SpawnOptions | undefined,
+  command: string,
+  ...args: string[]
+): Promise<ICommandResult> {
+  return await new Promise(
+    (resolve: (res: ICommandResult) => void, reject: (e: Error) => void): void => {
+      let cmdOutput = "";
+      let cmdOutputIncludingStderr = "";
+      const formattedArgs: string = args.join(" ");
```

Comment:
JSON.stringify the arrary? Since args may contain space, args.join(" ") will impact troubleshooting since we don't know it's 1 arg with space, or 2 args